### PR TITLE
BC-538 - task item with edit action

### PR DESCRIPTION
**Pull request by odalys-dataport**
# Short Description

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 1]
```diff
@@ -1,106 +1,173 @@
 import Vuetify from "vuetify";
-
 import mocks from "@@/stories/mockData/Tasks";
-
 import vTaskItemTeacher from "./vTaskItemTeacher";
 
 const { tasksTeacher, drafts } = mocks;
+const routerPushStub = { push: jest.fn() };
+
+Object.defineProperty(window, "innerWidth", {
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:8]: Make it sense to put it to a function or move to the into the describe block?

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 2]
```diff
@@ -1,106 +1,173 @@
...
+		propsData: props,
+		attachTo: document.body,
+		...options,
+	});
+};
 
 describe("@components/molecules/vTaskItemTeacher", () => {
-	let vuetify;
+	// eslint-disable-next-line no-unused-vars
+	let vuetify; // TODO - figure this out
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:30]: What should we du with this 2 lines? 
The var is in used in line 33.

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 3]
```diff
@@ -1,106 +1,173 @@
...
-			vuetify,
-			propsData: {
+	describe("course name", () => {
+		it("Should render subtitle with course name and no due date for tasks without due date", () => {
+			const wrapper = getWrapper({
 				task: tasksTeacher[7],
-			},
+			});
+
+			expect(wrapper.find(".v-list-item__subtitle").html()).toMatchSnapshot();
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:63]: toMatchSnapshot(); for a component unit test looks wrong. Same in the next tests.

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 4]
```diff
@@ -1,106 +1,173 @@
...
-			},
+			});
+
+			expect(wrapper.text()).toStrictEqual(
+				expect.not.stringContaining("Thema ")
+			);
 		});
 
-		expect(wrapper.find(".v-list-item__subtitle").html()).toMatchSnapshot();
+		it.todo("missing complete coverage for coursename and topic");
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:94]: Great than we need tests for it. :D

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 5]
```diff
@@ -1,106 +1,173 @@
...
 	});
 
-	it("accepts valid task props", () => {
-		const { validator } = vTaskItemTeacher.props.task;
-		const validTasks = tasksTeacher;
+	describe("when menu is used", () => {
+		describe("mouse events", () => {
+			it.todo("should show menu btn on hover");
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:99]: Is it possible to test it.

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 6]
```diff
@@ -1,106 +1,173 @@
...
-			propsData: {
+	describe("topic", () => {
+		it("should display topic", () => {
+			const wrapper = getWrapper({
 				task: drafts[1],
-			},
+			});
+
+			expect(wrapper.text()).toStrictEqual(
+				expect.not.stringContaining("Thema ")
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:90]: toStrictEqual in combination with expect.not looks wrong.

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 7]
```diff
@@ -1,106 +1,173 @@
...
+
+				const menuBtn = wrapper.find(".v-btn");
+				await menuBtn.trigger("click");
+				await menuBtn.trigger("click");
+
+				expect(wrapper.vm.isMenuActive).toBe(false);
+				expect(wrapper.find(".menuable__content__active").exists()).toBe(false);
+			});
+
+			it.todo("should close & hide menu on outside click");
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:127]: todo for this pr?

File: src/components/molecules/vTaskItemTeacher.unit.js [hunk 8]
```diff
@@ -1,106 +1,173 @@
...
+			const editBtn = wrapper.find(".task-action");
+
+			expect(editBtn.attributes("href")).toBe(
+				`/homework/${tasksTeacher[0].id}/edit`
+			);
+		});
 
-	it.todo("missing complete coverage for coursename and topic");
+		it("always show menu on mobile", () => {
+			Object.defineProperty(window, "innerWidth", {
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.unit.js:158]: We use this notation on 2 places in this file, moving to a function looks usefull.

File: src/components/molecules/vTaskItemTeacher.vue [hunk 9]
```diff
@@ -1,64 +1,111 @@
...
+					<v-list-item-subtitle>{{
+						$t("components.molecules.VTaskItemTeacher.graded")
+					}}</v-list-item-subtitle>
+					<v-list-item-title data-testid="taskGraded">{{
+						task.status.graded
+					}}</v-list-item-title>
+				</v-list-item-action>
+			</section>
+			<v-list-item-action :id="`task-menu-${task.id}`" class="context-menu-btn">
+				<v-menu
```
Comment (CeEv) [inline src/components/molecules/vTaskItemTeacher.vue:62]: v-menu looks like it make sense to move it in additional component.

File: tests/unit/__snapshots__/components/molecules/vTaskItemTeacher.unit.js.snap [hunk 10]
```diff
@@ -17,3 +17,21 @@ exports[`@components/molecules/vTaskItemTeacher Should render subtitle with no c
   &nbsp;– Kein Abgabedatum
 </div>
 `;
+
+exports[`@components/molecules/vTaskItemTeacher course name Should render subtitle with course name and due date 1`] = `
```
Comment (CeEv) [inline tests/unit/__snapshots__/components/molecules/vTaskItemTeacher.unit.js.snap:21]: 3 times 'S'hould instant of 's'. but this is in the __snapshots__ i am not sure if it make sense to fix this.

**Review by CeEv — CHANGES_REQUESTED** [review]

File: src/components/molecules/vTaskItemTeacher.unit.js (hunk 3 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.unit.js:63]: These are still old tests, I just moved them around a little. We should rework this definitely, but not in the scope of this ticket.

File: src/components/molecules/vTaskItemTeacher.unit.js (hunk 4 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.unit.js:94]: Also part of the old code and would have been part of the general task refactor

File: src/components/molecules/vTaskItemTeacher.unit.js (hunk 5 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.unit.js:99]: We have tried and couldn't get it to work. It's manually tested and tested in e2e test, so we put it as a todo until we've been hit with inspiration on how to fix it.

File: src/components/molecules/vTaskItemTeacher.unit.js (hunk 6 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.unit.js:90]: Also part of the old code

File: src/components/molecules/vTaskItemTeacher.unit.js (hunk 7 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.unit.js:127]: Same as the hover effect. Marked it as a todo for now, so we can finish this PR in time

File: src/components/molecules/vTaskItemTeacher.unit.js (hunk 8 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.unit.js:158]: I'll look into it

File: src/components/molecules/vTaskItemTeacher.vue (hunk 9 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.vue:62]: Definitely, I plan to do this within 1071, where more actions are added.

File: src/components/molecules/vTaskItemTeacher.unit.js (hunk 2 above)
Comment (odalys-dataport) [reply src/components/molecules/vTaskItemTeacher.unit.js:30]: I'll look into it

**Review by CeEv — APPROVED** [review]

**Review by CeEv — APPROVED** [review]

**Pull request merged**